### PR TITLE
frontend: common: Add ConditionList, PhaseLabel and getReadyCondition

### LINKED ISSUE #5928: feat: shared React component library for Headlamp plugin authors
## Is your feature request related to a problem? Please describe the impact that the lack of the feature requested is creating.

Every Headlamp plugin that surfaces Kubernetes custom resources faces the same set of UI problems: rendering `.status.conditions[]` arrays, building consistent detail page layouts, displaying phase strings as colored chips, and handling loading, error, and empty states. Right now each plugin author solves these independently, which means duplicated code, inconsistent UX across plugins, and a steeper learning curve for new contributors who have to reverse-engineer patterns from existing plugins instead of reaching for a documented, tested component.

With multiple plugins either recently merged or actively in development (Agones, Argo CD, Kueue, Metal3, Tinkerbell, Cluster API, Kubeflow, Strimzi, Knative), this duplication is becoming a real cost. There is no shared foundation that plugin authors can build on.

## Describe the solution you'd like

A shared component library scoped specifically to Headlamp plugin development, either living under `frontend/src/plugin-components` in the main repo or published as a standalone package under the `@headlamp-k8s` namespace.

The initial set of components I am proposing, based on patterns I have seen repeated across the existing plugin PRs:

- `<CRDStatusBadge />` - renders a Kubernetes `.status.conditions[]` array with type, status, and reason surfaced consistently
- `<ResourceDetailLayout />` - a standard scaffold for custom resource detail pages with metadata, status, and events sections
- `<CRDTable />` - a table component wired for custom resources with sorting, filtering, and pagination
- `<ConditionList />` - renders condition arrays with icons and color coding tied to condition status
- `<PhaseChip />` - maps any `.status.phase` string to a MUI chip with sensible color defaults for common Kubernetes phase values

Each component would ship with:
- Full TypeScript types
- Unit tests
- Storybook stories for visual reference and manual testing
- JSDoc on all props

## What users will benefit from this feature?

Plugin developers primarily. Anyone building a Headlamp plugin that surfaces custom resources would have a set of tested, documented building blocks instead of building from scratch. This also benefits end users indirectly because consistent components mean a consistent UX across all plugins, so switching between the Agones plugin and the Argo CD plugin does not feel like a different product.

## Are you able to implement this feature?

Yes (I will propose a PR).

Before opening a PR I wanted to raise this here first to get maintainer input on two things:

1. Whether this belongs in the main repo or as a separate package under `@headlamp-k8s`
2. Whether there are existing internal patterns or component abstractions I should align with before starting

Happy to jump on the Slack channel to discuss further if that is easier.

## Additional context

I have hands-on experience with the kubernetes-sigs/headlamp codebase across multiple layers and this proposal comes directly from noticing the same patterns being re-implemented across plugin PRs.

### PR DESCRIPTION
**Pull request by Athang69**
## Summary

This PR reduces UI code duplication across the codebase by introducing two shared components and one utility function in `frontend/src/components/common/`.

## Related Issue

Fixes #5928

## Background

While auditing the codebase as part of issue #5928, I found three concrete duplication patterns:

1. `.status.conditions[]` rendering is implemented four different ways across `Resource.tsx`, `gateway/ClassList.tsx`, `gateway/GatewayDetails.tsx`, and `crd/CustomResourceDetails.tsx`, each with different logic for deriving the status colour.

2. `.status.phase` to `StatusLabel` mapping is done inline in `namespace/Details.tsx`, `namespace/List.tsx`, `portforward/index.tsx`, and via a storage-specific `StatusLabelByPhase` helper in `storage/utils.tsx`.

3. `conditions.find(c => c.type === 'Ready')` is duplicated at least 9 times across `pod/List.tsx`, `cluster/Charts/StatusCharts.tsx`, `diagnostics/Diagnostics.tsx`, `resourceMap/nodes/KubeObjectStatus.tsx`, `node/Details.tsx`, and `node/List.tsx`.

## Changes

- Added `ConditionList` component in `common/ConditionList.tsx` that accepts a `KubeCondition[]` and renders a consistent table with type, status, last transition, and reason columns. Status colour is derived uniformly: `True` -> success, `False` -> error, anything else -> neutral.
- Added `getReadyCondition` utility exported from `ConditionList.tsx` that centralises the `conditions.find(c => c.type === 'Ready')` pattern duplicated nine times across the codebase.
- Added `PhaseLabel` component in `common/PhaseLabel.tsx` that maps any `.status.phase` string to a coloured `StatusLabel` via configurable `successPhase` (default `'Active'`) and `warningPhases` props.
- Exported both new components from `common/index.ts` so plugin authors and internal callers can import them from the common barrel.
- Migrated `namespace/Details.tsx` to use `PhaseLabel` in place of its local `makeStatusLabel` function.
- Migrated `storage/utils.tsx` `StatusLabelByPhase` to delegate to `PhaseLabel`, marked it `@deprecated`, and removed the now-unused `t` argument from its signature and the two call sites in `ClaimDetails.tsx` and `VolumeDetails.tsx`.

## Steps to Test

1. Navigate to any Namespace detail page and confirm the status chip renders green for Active and red for Terminating.
2. Navigate to any PersistentVolume detail page and confirm Bound renders green and Available renders yellow.
3. Navigate to any PersistentVolumeClaim detail page and confirm the phase chip renders correctly.
4. Navigate to any CRD custom resource detail page that has conditions and confirm the conditions table renders as before.

## Notes for the Reviewer

- The follow-up callers of `ConditionList` (`gateway/ClassList.tsx`, `gateway/GatewayDetails.tsx`) and the nine `getReadyCondition` call sites are intentionally left for a follow-up to keep this PR focused and reviewable. Happy to do that in a separate PR or in this one if preferred.
- `npx tsc --noEmit` reports 4 pre-existing errors in `ReleaseNotesModal.tsx` and `patchUtils.ts` (missing type declarations for `remark-gfm` and `fast-json-patch`). These exist on `main` before this PR and are unrelated to these changes.

### CODE REVIEW

File: frontend/src/components/storage/utils.tsx [hunk 1]
```diff
@@ -14,23 +14,13 @@
...
-    >
-      {phaseMap[phase] || phase}
-    </StatusLabel>
-  );
+/**
+ * @deprecated Use PhaseLabel from common/PhaseLabel directly.
+ * Kept for backwards compatibility with existing storage list/details callers.
+ */
+export function StatusLabelByPhase(phase: string) {
+  return <PhaseLabel phase={phase} successPhase="Bound" warningPhases={['Available']} />;
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:26]: The t argument has been removed from StatusLabelByPhase's signature and both call sites (ClaimDetails.tsx and VolumeDetails.tsx) have been updated accordingly. No callers pass t anymore so there is no breakage.

[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:26]: The t argument has been removed from StatusLabelByPhase's signature and both call sites (ClaimDetails.tsx and VolumeDetails.tsx) have been updated accordingly. No callers pass t anymore so there is no breakage.

File: frontend/src/components/storage/VolumeDetails.tsx [hunk 2]
```diff
@@ -22,9 +22,9 @@ import Link from '../common/Link';
 import { DetailsGrid } from '../common/Resource';
 import { StatusLabelByPhase } from './utils';
 
-export function makePVStatusLabel(item: PersistentVolume, t: (key: string) => string) {
+export function makePVStatusLabel(item: PersistentVolume) {
   const status = item.status?.phase ?? '';
-  return StatusLabelByPhase(status, t);
+  return StatusLabelByPhase(status);
 }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/VolumeDetails.tsx:28]: This is a diff display artifact. The file contains only one return statement: `return StatusLabelByPhase(status)`.

File: frontend/src/components/common/PhaseLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * <PhaseLabel phase={pf.status} successPhase={PORT_FORWARD_RUNNING_STATUS} />
+ */
+export function PhaseLabel({
+  phase,
+  successPhase = 'Active',
+  warningPhases = [],
+}: PhaseLabelProps) {
+  if (phase === undefined || phase === null) {
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/PhaseLabel.tsx]: Intentional. Returning null when phase is absent lets callers pass resource.status?.phase directly without needing to guard at the call site. This matches the pattern used by other nullable-safe components in the codebase.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for this PR.


There are snapshot mismatches in CI. You can fix them locally by running `cd frontend && npm run test -- -u`.

<details>
<summary>How to update snapshots</summary>

Run `cd frontend && npm run test -- -u` to regenerate all snapshots. Review the diff to make sure the visual changes are intentional, then commit the updated snapshot files.

</details>

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for this PR.


The open review comments from Copilot still need attention — can you have a look? Once addressed, please mark them as resolved.

File: frontend/src/components/common/PhaseLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { StatusLabel, StatusLabelProps } from './Label';
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/PhaseLabel.tsx:18]: Other files in this PR (ConditionList.tsx, storage/utils.tsx) include this import and pass ESLint without issue. Keeping it for consistency.

File: frontend/src/components/storage/utils.tsx [hunk 5]
```diff
@@ -14,23 +14,13 @@
...
 }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:27]: Acknowledged. The raw phase string is intentional here - StatusLabelByPhase is marked deprecated and downstream callers (VolumeList, ClaimList) can migrate to PhaseLabel directly and add their own translation wrapper if needed.

File: frontend/src/components/storage/utils.tsx [hunk 6]
```diff
@@ -14,23 +14,14 @@
...
-      {phaseMap[phase] || phase}
-    </StatusLabel>
-  );
+/**
+ * @deprecated Use PhaseLabel from common/PhaseLabel directly.
+ * Note: the t translation argument has been removed. Callers
+ * that previously passed t as a second argument must be updated.
+ */
+export function StatusLabelByPhase(phase: string) {
+  return <PhaseLabel phase={phase} successPhase="Bound" warningPhases={['Available']} />;
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:26]: Acknowledged. StatusLabelByPhase is deprecated and its callers (VolumeList, ClaimList) can add translation at their own call site when migrating to PhaseLabel directly. Keeping as-is.

[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:26]: All call sites (ClaimList.tsx, VolumeDetails.tsx, VolumeList.tsx, ClaimDetails.tsx) have already been updated in this PR to drop the t argument. There are no remaining callers passing t.

File: frontend/src/components/common/PhaseLabel.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  if (phase === successPhase) {
+    status = 'success';
+  } else if (warningPhases.includes(phase)) {
+    status = 'warning';
+  } else {
+    status = 'error';
+  }
+
+  return <StatusLabel status={status}>{phase}</StatusLabel>;
+}
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/PhaseLabel.tsx:81]: Will add unit tests for PhaseLabel, ConditionList status color rules, and getReadyCondition in a follow-up PR to keep this PR reviewable.

File: frontend/src/components/storage/utils.tsx [hunk 8]
```diff
@@ -14,23 +14,14 @@
...
 }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:27]: Acknowledged. The i18n mapping is intentional to drop as StatusLabelByPhase is deprecated. Callers migrating to PhaseLabel directly can wrap phase translation at their own call site.

File: frontend/src/components/common/PhaseLabel.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/PhaseLabel.tsx:81]: Will add unit tests for PhaseLabel in a follow-up PR to keep this focused.

File: frontend/src/components/common/ConditionList.tsx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+ * appears at least 9 times across the codebase. Centralising it here
+ * makes call-sites shorter and keeps the semantics in one place.
+ *
+ * @param conditions - Array of KubeCondition objects to search.
+ */
+export function getReadyCondition(
+  conditions: KubeCondition[] | undefined | null
+): KubeCondition | undefined {
+  return conditions?.find(c => c.type === 'Ready');
+}
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/ConditionList.tsx:38]: Will add unit tests for getReadyCondition in a follow-up PR.

File: frontend/src/components/common/ConditionList.tsx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+  if (!conditions || conditions.length === 0) {
+    return null;
+  }
+
+  function statusFromCondition(condition: KubeCondition): StatusLabelProps['status'] {
+    if (condition.status === 'True') return 'success';
+    if (condition.status === 'False') return 'error';
+    return '';
+  }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/ConditionList.tsx:84]: '' is an explicitly valid value in StatusLabelProps['status'] (defined as 'success' | 'warning' | 'error' | ''). It renders a neutral chip, which is the correct behaviour for conditions with status other than True or False.

File: frontend/src/components/storage/utils.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/storage/utils.tsx:27]: All call sites in this PR (ClaimList.tsx, VolumeDetails.tsx, VolumeList.tsx, ClaimDetails.tsx) have already been updated to drop the t argument. There are no remaining callers passing t so no breaking change occurs in practice.

File: frontend/src/components/common/ConditionList.tsx [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+ * @example
+ * // With the extra Last Update column
+ * <ConditionList conditions={resource.status?.conditions} showLastUpdate />
+ */
+export function ConditionList({ conditions, showLastUpdate = false }: ConditionListProps) {
+  const { t } = useTranslation(['glossary', 'translation']);
+
+  if (!conditions || conditions.length === 0) {
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/ConditionList.tsx:78]: Updated the doc comment to say "Renders nothing when null, undefined, or empty"

File: frontend/src/components/common/PhaseLabel.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Athang69) [reply frontend/src/components/common/PhaseLabel.tsx:81]: Will add unit tests covering null/undefined/'', successPhase match, warningPhases match, and default error fallback in a follow-up PR to keep this PR focused.

**Comment by Athang69** [comment]
@illume, the open Copilot comments are now addressed. The remaining ones are suggestions to add unit tests for PhaseLabel, ConditionList, and getReadyCondition - I've acknowledged those and will follow up in a separate PR to keep this one focused. The snapshot is also fixed. PTAL when convenient, the PR is ready for review.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for these changes.


I noticed the GitHub CI frontend i18n check is failing. It looks like the translation files need to be updated. You can run `cd frontend && npm run i18n` locally to update them and then commit the result.

<details>
<summary>How to update translation files</summary>

Run `cd frontend && npm run i18n` to regenerate the translation files. Commit the updated files alongside your other changes.

</details>

**Comment by Athang69** [comment]
> Thanks for these changes.
> 
> I noticed the GitHub CI frontend i18n check is failing. It looks like the translation files need to be updated. You can run `cd frontend && npm run i18n` locally to update them and then commit the result.
> 
> How to update translation files

@illume I've fixed the tests and addressed copilot's reviews. PTAL.
Thankss!

**Review by illume — APPROVED** [review]
🎉 thanks!
